kamon-core/src/main/scala/kamon/trace/MetricsOnlyContext.scala

```diff
@@ -52,11 +54,21 @@ private[kamon] class MetricsOnlyContext(traceName: String, val token: String, iz
     val traceElapsedTime = NanoInterval.since(startTimestamp)
     _elapsedTime = traceElapsedTime
 
-    if (Kamon.metrics.shouldTrack(name, TraceMetrics.category))
-      Kamon.metrics.entity(TraceMetrics, name).elapsedTime.record(traceElapsedTime.nanos)
+    if (Kamon.metrics.shouldTrack(name, TraceMetrics.category)) {
+      val traceEntity = Kamon.metrics.entity(TraceMetrics, name)
```

Comment:
we should get the tags PR merged first, then rebase this branch and ensure that this uses tags as well.